Update to plot_confusion_matrix (figsize argument and to work if Seaborn is used)

frankherfert opened the pull request:
Using the confusion matrix in a jupyter notebook returns a plot that is quite small. If Seaborn is also used, some values in the plot are hard to read (white text on white lines).

I have added a figsize-argument to the plot_confusion_matrix and changed the way that values are displayed (now with neutral background box). For larger plots all text-elements scale with the figsize-argument.

reiinakano commented:
Hi @frankherfert , first thanks a lot for taking the time to submit this PR! I really appreciate it.

I'll admit I never actually tried using the library on Jupyter. After seeing this, I tried to generate the confusion matrix on Jupyter and you're right, it does look a bit small and weird.

Having said that though, I tested your PR on my local workstation and this is the image that was generated. 

![11](https://cloud.githubusercontent.com/assets/18363734/23303247/e9cdc9dc-facd-11e6-8c90-4aeb72a39f11.png)

I don't think this is how you wanted it to look?

frankherfert commented:
Hi @reiinakano 

thanks for reviewing my suggestions.
I tried the confusion matrix on a binary classifier and the boxes looked better than on a 10x10 matrix =).

I removed the boxes and changed the default font-sizes to "medium" and "large" as you suggested. Up to figsize=10 this looks ok but above that the user would need to change this manually:
![screenshot from 2017-02-24 15-23-48](https://cloud.githubusercontent.com/assets/15942370/23306789/4d8d3a76-faa5-11e6-86a4-779728d781e6.png).

If the additional parameters are useful, I can work them into your other plots as well.

reiinakano commented:
Yes, I totally agree that the natural way forward is to incorporate these extra parameters into all the other plots.

Anyway, the reason I don't want to incorporate autoscaling of font sizes with the `figsize` is different datasets and different people will have different ideas of what the "best" sizes are. The safest way I think is to stay with the Matplotlib defaults, which is to keep fontsizes the same regardless of `figsize` and let the user adjust as he/she pleases.

Anyway, I'm just about ready to merge. Instead of `figsize` taking an int, could you change it to a 2-tuple so that width and height can be selected as well e.g. (6, 6). Also, keep the Matplotlib default of `None` for `figsize`.

reiinakano commented on scikitplot/classifiers.py:
```diff
@@ -54,7 +54,7 @@ def classifier_factory(clf):
 
 def plot_confusion_matrix(clf, X, y, title=None, normalize=False, do_cv=True, cv=None,
-                          shuffle=True, random_state=None, ax=None):
+                          shuffle=True, random_state=None, ax=None, figsize=6, title_fontsize="large", text_fontsize="medium"):
```
style fix: Put the extra arguments on a new line

```python
 def plot_confusion_matrix(clf, X, y, title=None, normalize=False, do_cv=True, cv=None,
                           shuffle=True, random_state=None, ax=None, figsize=6, 
                           title_fontsize="large", text_fontsize="medium"):
```

reiinakano commented on scikitplot/plotters.py:
```diff
@@ -22,7 +22,7 @@
 
 
-def plot_confusion_matrix(y_true, y_pred, title=None, normalize=False, ax=None):
+def plot_confusion_matrix(y_true, y_pred, title=None, normalize=False, ax=None, figsize=6, title_fontsize="large", text_fontsize="medium"):
```
style fix: Put the extra arguments on a new line

reiinakano commented on scikitplot/classifiers.py:
```diff
@@ -100,6 +100,15 @@ def plot_confusion_matrix(clf, X, y, title=None, normalize=False, do_cv=True, cv
         ax (:class:`matplotlib.axes.Axes`, optional): The axes upon which to plot
             the learning curve. If None, the plot is drawn on a new set of axes.
 
+        figsize (int, optional): Rectangular figure size of the plot. Defaults to 6.
```
accept 2-tuple instead of integer

reiinakano commented:
It turns out I can commit changes to your PR, so since I had downtime and the fixes were easy, I just did it myself. :) 

Thanks a lot @frankherfert for the contribution! This will surely be very useful for Jupyter notebook users.

frankherfert commented:
Thanks for finalizing it. I had some downtime until now.

I tried to use a tuple for the figsize but anything beside two identical numbers produced weird looking plots, which is why I chose just one int.

reiinakano commented:
Ahh, I see. Though I'm planning to add the new arguments to every other plot anyway, and non-square figsizes *might* make sense for those plots. Best to have a uniform signature for them all and be prepared for the future.
